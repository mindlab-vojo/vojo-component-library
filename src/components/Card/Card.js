import React from 'react'
import PropTypes from 'prop-types'

import * as Card from './Card.style'

function CardComponent({ 
  backgroundColor,
  children,
  hasPointer,
  onClickCard,
  padding,
}) {
  return (
    <Card.Card 
      backgroundColor={backgroundColor}
      className="Card"
      hasPointer={hasPointer}
      onClick={onClickCard}
      padding={padding}>
      {children}
    </Card.Card>
  )
}

CardComponent.propTypes = {
  backgroundColor: props.string,
  children: PropTypes.node,
  hasPointer: PropTypes.bool,
  onClickCard: PropTypes.func,
  padding: PropTypes.string,
}

CardComponent.defaultProps = {
  padding: "14px 10px"
}

export default CardComponent

