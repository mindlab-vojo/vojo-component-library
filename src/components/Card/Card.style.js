import styled from 'styled-components'
import style from '../../style'

export const Wrapper = styled.div`
    border: 1px solid ${style.colors.greyColor};
    border-radius: 8px;
    display: flex;
    flex-direction: column;
    height: auto;
    overflow: hidden;
    width: 100%;

    ${
  style.breakpoints.largerThanTablet`
                height: 400px;
                width: 250px
            `
}
`

export const Image = styled.div`
    background-image: url(${props => props.src});
    background-position: center center;
    background-size: cover;
    height: 150px;
    width: 100%;
`

export const UpperBody = styled.div`
    display: flex;
    width: 100%;
`

export const LowerBody = styled.div`
    display: flex;
    flex-direction: column;
    padding: 15px 15px;
`

export const Title = styled.h1`
    color: ${style.colors.primaryColor};
    font-size: 24px;
    margin: 0 0 10px;
`

export const Description = styled.p`
    margin: 0;
    width: 100%
`
