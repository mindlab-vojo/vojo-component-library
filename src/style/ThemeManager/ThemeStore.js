import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore