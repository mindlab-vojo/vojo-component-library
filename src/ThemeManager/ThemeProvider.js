import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer