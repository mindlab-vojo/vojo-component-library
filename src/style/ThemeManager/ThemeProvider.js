import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer