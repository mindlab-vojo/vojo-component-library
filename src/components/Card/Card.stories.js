import React from 'react';
import Card from '.';
import { withKnobs, text, boolean } from "@storybook/addon-knobs";
import { withA11y } from '@storybook/addon-a11y';

import { ThemeProvider } from './../../ThemeManager/ThemeProvider'
import { redVojo } from './../../style/themes'

export default {
  title: 'Card',
  component: Card,
  decorators: [withKnobs, withA11y]
}

export const component = () => {
  const description = text("description", "It's difficult to find examples of lorem ipsum in use before Letraset made it popular as a dummy text in the 1960s, although McClintock says he remembers coming across the lorem ipsum passage in a book of old metal type samples.")
  const image = text("image", "https://lastfm.freetls.fastly.net/i/u/770x0/5721b67079174fcccfe65965458b2559.jpg")
  const title = text("title", "Card Sample")
  const redTheme = boolean("Red Themed", false)
  return (
    <ThemeProvider theme={redTheme ? redVojo : null}>
      <div>
        <Card 
          description={description} 
          image={image}
          title={title}
        />
      </div>
    </ThemeProvider>
  )
}

component.story = {
  name: 'Card',
};
  