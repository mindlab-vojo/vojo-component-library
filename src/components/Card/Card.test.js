import React from 'react';
import { render } from '@testing-library/react';

import Card from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test Card component render', () => {
  const component = render(
    <ThemeProvider>
      <Card/>
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
