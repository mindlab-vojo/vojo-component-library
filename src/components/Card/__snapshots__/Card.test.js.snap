// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm kvTteP Card__Wrapper"
      >
        <div
          class="sc-AxiKw eIMRRL Card__Wrapper__UpperBody"
        >
          <div
            class="sc-AxirZ jZFMiW Card__Wrapper__UpperBody__Image"
            src="https://miro.medium.com/max/4240/0*6lzpRfcdt7wU0HAN."
          />
        </div>
        <div
          class="sc-AxhCb kwcBJT Card__Wrapper__LowerBody"
        >
          <h1
            class="sc-AxhUy fTTxeE Card__Wrapper__LowerBody__Title"
          >
            Testing
          </h1>
          <p
            class="sc-AxgMl eErQtE Card__Wrapper__LowerBody__Description"
          >
            Testing this component with props
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm kvTteP Card__Wrapper"
    >
      <div
        class="sc-AxiKw eIMRRL Card__Wrapper__UpperBody"
      >
        <div
          class="sc-AxirZ jZFMiW Card__Wrapper__UpperBody__Image"
          src="https://miro.medium.com/max/4240/0*6lzpRfcdt7wU0HAN."
        />
      </div>
      <div
        class="sc-AxhCb kwcBJT Card__Wrapper__LowerBody"
      >
        <h1
          class="sc-AxhUy fTTxeE Card__Wrapper__LowerBody__Title"
        >
          Testing
        </h1>
        <p
          class="sc-AxgMl eErQtE Card__Wrapper__LowerBody__Description"
        >
          Testing this component with props
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
